### PR TITLE
[ON HOLD] PIM-6037: Prevent association code to be an integer

### PR DESCRIPTION
**Pull request by damien-carcel**
**Description**

**Definition Of Done**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | :negative_squared_cross_mark:
| Added Behats                      | :white_check_mark:
| Changelog updated                 | :white_check_mark:
| Review and 2 GTM                  | :clock1:
| Micro Demo to the PO (Story only) | :clock1:
| Migration script                  | :negative_squared_cross_mark:
| Tech Doc                          | :negative_squared_cross_mark:

### CODE REVIEW

File: src/Pim/Component/Catalog/Updater/Setter/AssociationFieldSetter.php [hunk 1]
```diff
@@ -98,6 +98,7 @@ protected function addMissingAssociations(ProductInterface $product)
      * Set products and groups to associations
      *
      * @param ProductInterface $product
+     * @param                  $data
```
Comment (solivier) [inline src/Pim/Component/Catalog/Updater/Setter/AssociationFieldSetter.php]: missing type

Comment (a2xchip) [inline src/Pim/Component/Catalog/Updater/Setter/AssociationFieldSetter.php]: :+1:

**Comment by damien-carcel** [comment]
Another way to fix it is to change the standard format:
- before
```json
{
   "XSELL": {
       "groups": ["group1", "group2"],
       "products": ["AKN_TS1", "AKN_TSH2"]
   },
   "UPSELL": {
       "groups": ["group3", "group4"],
       "products": ["AKN_TS3", "AKN_TSH4"]
   }
}
```
- after
```json
{
   {
       "code": "XSELL",
       "groups": ["group1", "group2"],
       "products": ["AKN_TS1", "AKN_TSH2"]
   },
   {
       "code": "UPSELL",
       "groups": ["group3", "group4"],
       "products": ["AKN_TS3", "AKN_TSH4"]
   }
}
```

This is not a trivial change, and need an update of the RFC and an approval of it.

So PR is on hold for now.

**Comment by nidup** [comment]
Still useful PR?

**Comment by damien-carcel** [comment]
Usefull, I don't know :). I was keeping it open because the corresponding SDS ticket is still open.
But in fact I can close it and open it again if ever needed!

**Pull request closed without merging**